doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+
+Rough point estimate for this story is 8 points.
+#### Story 3 - Common Validation
+The scorecard would use a common validation codebase to verify bundle contents. This story supports a single “source of truth” and avoids duplication or variance of validation logic. The scorecard user would be able to specify a bundle on the command line, that bundle would then be validated.  For example:
+ * operator-sdk scorecard --bundle ./my-operator
+
+
+Tasks for this story include:
+ * determine the correct validation library to use
+ * add support for a bundle directory flag to the CLI
+ * change the scorecard code to use that validation library when validating a CR, CSV, or given a bundle directory
+ * format the validation output to present to the CLI user in text or json
+
+Rough point estimate in this story is 5 points.
+
+#### Story 4 - Update scorecard Documentation
+Tasks:
+ * Document changes to CLI
+ * Document new output formats
+ * Document changes to configuration
+ * Document breaking changes and removals
+
+
+
+### Implementation Details/Notes/Constraints
+
+A scheme for applying the labels to the tests would need to be developed as well as label names to categorize tests.
+
+The “source of truth” for validation would need to be established such as which library the scorecard would use to perform CR/CSV/bundle validation.
+
+If no runtime tests are specified, then the scorecard would only run the static tests and not depend on a running cluster.
+
+### Risks and Mitigations
+
+The scorecard would implement a version flag in the CLI to allow users to migrate from current functionality to the proposed functionality (e.g. v1alpha2):
+ * operator-sdk scorecard --version v1alpha2
+
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+maturity levels.
+
+##### Dev Preview -> Tech Preview
```

Comment:
IMHO this doc recommends we change the features of the scorecard, so:
1. I'd adopted the `incremental development process` where each deliverable change with its docs would be merge in the master.  (In this way we can add value to the users and if the scope change we will not require to start from the scratch / agile process  )
2. In the end, if we still think that it should be another tool and etc .. we could extract the impl for a new  repo/project